juju/juju
# Merge 2.9

Merge 2.9

#14049 [JUJU-1129] Refactor resource client api to be consistent with other juju client api
#14051 [JUJU-1125] Improve error message for local OCI image resource
#14048 [JUJU-1119] work around golang http proxy caching by default - add more logging
#14055 Added two examples to juju status help
#14058 [JUJU-1129] Refactor payload client facades to match standard practice
#14059 Update add-unit help to clarify lp#1971956
#14060 Increase the number of open file descriptors
#14057 [JUJU-1157] Check that an AgentStreamKey value exists
#14062 Remove the componentisation of resources and payloads
#14054 Update debug-log help text
#14064  [JUJU-1159] Improved help model-defaults message
#14066 Bump actions/upload-artifact from 3.0.0 to 3.1.0
#14070 [JUJU-1129] Move resources files to a single top level package
#14071 Update the model-defaults help text

Conflicts were charm and resource import paths, deleted resource files, and struct initialisations.
```
# Conflicts:
#       api/client/resources/client.go
#       api/client/resources/client/base_test.go
#       api/client/resources/client/client_upload_test.go
#       apiserver/facades/controller/caasapplicationprovisioner/provisioner_test.go
#       caas/kubernetes/provider/k8s.go
#       caas/kubernetes/provider/k8s_test.go
#       cmd/containeragent/initialize/package_test.go
#       cmd/containeragent/unit/package_test.go
#       cmd/juju/application/bundle_test.go
#       cmd/juju/application/deployer/bundlehandler.go
#       cmd/juju/application/deployer/deployer.go
#       cmd/juju/application/refresh.go
#       cmd/juju/commands/main.go
#       cmd/juju/commands/main_test.go
#       core/watcher/package_test.go
#       go.sum
#       payload/api/private/helpers_test.go
#       payload/context/base_test.go
#       payload/context/register_test.go
#       payload/context/utils.go
#       resource/charmhub.go
#       resource/charmstore.go
#       resource/context/context.go
#       resource/context/internal/content.go
#       resource/context/internal/content_test.go
#       resource/context/internal/resourcedir.go
#       resource/context/internal/stub_test.go
#       resource/context/utils.go
#       resource/context/utils_test.go
#       resource/repositories/mocks/mocks.go
#       resource/repositories/operations.go
#       resource/repositories/operations_test.go
#       resource/repositories/repository.go
#       resource/resource.go
#       resource/resourceadapters/interfaces.go
#       resource/resourceadapters/mocks/opener_mock.go
#       resource/resourceadapters/opener.go
#       resource/unit.go
#       snap/snapcraft.yaml
#       state/mocks/resources_mock.go
#       worker/caasapplicationprovisioner/application_test.go
#       worker/uniter/manifold.go
#       worker/uniter/runner/context/contextfactory.go
#       worker/uniter/runner/jujuc/mocks/context_mock.go
#       worker/uniter/runner/jujuc/payload-register.go
#       worker/uniter/runner/jujuc/restricted.go
#       worker/uniter/runner/jujuc/server.go
#       worker/uniter/uniter.go
#       worker/uniter/uniter_test.go
#       worker/uniter/util_test.go
```
## QA steps

See PRs


[JUJU-1129]: https://warthogs.atlassian.net/browse/JUJU-1129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1125]: https://warthogs.atlassian.net/browse/JUJU-1125?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1119]: https://warthogs.atlassian.net/browse/JUJU-1119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1129]: https://warthogs.atlassian.net/browse/JUJU-1129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1157]: https://warthogs.atlassian.net/browse/JUJU-1157?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1159]: https://warthogs.atlassian.net/browse/JUJU-1159?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1129]: https://warthogs.atlassian.net/browse/JUJU-1129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
